src/EditorFeatures/Core.Wpf/Interactive/InteractiveEvaluator.cs

```diff
@@ -314,18 +314,21 @@ private void LanguageBufferContentTypeChanged(object sender, ContentTypeChangedE
             // We're switching between the target language and the Interactive Command "language".
             // First, remove the current submission from the solution.
 
+            var documentId = _workspace.GetDocumentIdInCurrentContext(buffer.AsTextContainer());
             var oldSolution = _workspace.CurrentSolution;
+            var relatedDocumentIds = oldSolution.GetRelatedDocumentIds(documentId);
+
             var newSolution = oldSolution;
 
-            foreach (var documentId in _workspace.GetRelatedDocumentIds(buffer.AsTextContainer()))
```

Comment:
I guess this is the one place I would have figured out we didn't mind to use the old method, since this is workspace manipulation itself...

Reply:
I still prefer to not implicitly access CurrentSolution in the middle of code that modifies the current solution.